File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/OSOutcomeEventsV2Repository.java
package com.onesignal.outcomes;

import com.onesignal.OSLogger;
import com.onesignal.OneSignalApiResponseHandler;
import com.onesignal.outcomes.domain.OutcomeEventsService;
import com.onesignal.outcomes.model.OSOutcomeEventParams;

import org.json.JSONException;
import org.json.JSONObject;

class OSOutcomeEventsV2Repository extends OSOutcomeEventsRepository {

    OSOutcomeEventsV2Repository(OSLogger logger, OSOutcomeEventsCache outcomeEventsCache, OutcomeEventsService outcomeEventsService) {
        super(logger, outcomeEventsCache, outcomeEventsService);
    }

    @Override
    public void requestMeasureOutcomeEvent(String appId, int deviceType, OSOutcomeEventParams event, OneSignalApiResponseHandler responseHandler) {
        try {
            JSONObject jsonObject = event.toJSONObject();
            jsonObject.put(APP_ID, appId);
            jsonObject.put(DEVICE_TYPE, deviceType);

            outcomeEventsService.sendOutcomeEvent(jsonObject, responseHandler);
        } catch (JSONException e) {
            logger.error("Generating indirect outcome:JSON Failed.", e);
        }
    }
}

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/data/OSOutcomeEventsV2Repository.kt
package com.onesignal.outcomes.data

import com.onesignal.OSLogger
import com.onesignal.OneSignalApiResponseHandler
import com.onesignal.outcomes.OSOutcomeConstants
import com.onesignal.outcomes.domain.OSOutcomeEventParams
import org.json.JSONException

internal class OSOutcomeEventsV2Repository(logger: OSLogger,
                                           outcomeEventsCache: OSOutcomeEventsCache,
                                           outcomeEventsService: OutcomeEventsService) : OSOutcomeEventsRepository(logger, outcomeEventsCache, outcomeEventsService) {
    override fun requestMeasureOutcomeEvent(appId: String, deviceType: Int, event: OSOutcomeEventParams, responseHandler: OneSignalApiResponseHandler) {
        try {
            event.toJSONObject()
                    .put(OSOutcomeConstants.APP_ID, appId)
                    .put(OSOutcomeConstants.DEVICE_TYPE, deviceType)
                    .also { jsonObject ->
                        outcomeEventsService.sendOutcomeEvent(jsonObject, responseHandler)
                    }
        } catch (e: JSONException) {
            logger.error("Generating indirect outcome:JSON Failed.", e)
        }
    }
}